test/adaptive_load/input_variable_setter_test.cc

```diff
@@ -0,0 +1,45 @@
+#include "external/envoy/source/common/config/utility.h"
+
+#include "adaptive_load/input_variable_setter_impl.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+
+TEST(RequestsPerSecondInputVariableSetterConfigFactoryTest, GeneratesEmptyConfigProto) {
+  InputVariableSetterConfigFactory& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<InputVariableSetterConfigFactory>(
+          "nighthawk.rps");
+
+  Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+
+  nighthawk::adaptive_load::RequestsPerSecondInputVariableSetterConfig expected_config;
+
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
+}
+
+TEST(RequestsPerSecondInputVariableSetterConfigFactoryTest, CreatesPlugin) {
+  nighthawk::adaptive_load::RequestsPerSecondInputVariableSetterConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+
+  InputVariableSetterConfigFactory& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<InputVariableSetterConfigFactory>(
+          "nighthawk.rps");
+  InputVariableSetterPtr plugin = config_factory.createInputVariableSetter(config_any);
+
+  EXPECT_NE(dynamic_cast<RequestsPerSecondInputVariableSetter*>(plugin.get()), nullptr);
```

Comment:
Maybe check the plugin's `name()` yields "nighthawk.rps"?

Reply:
Added another test for the factory name. I still want to test the final created class because the implementations of these factories are always copy-pasted from other factories that all return `InputVariableSetterPtr`.